java/de/yessoft/android/mohammad/ui/MainActivity/fragments/NewsFeedFragment.java

```diff
@@ -0,0 +1,36 @@
+package de.yessoft.android.mohammad.ui.MainActivity.fragments;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import de.yessoft.android.mohammad.R;
+import de.yessoft.android.mohammad.ui.MainActivity.fragments.NewsFeedRV.NewsFeedAdapter;
+
+public class NewsFeedFragment extends Fragment {
+    //private static final String TAG = "ML_NewsFeedFragment";
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_news_feed, container, false);
+
+        RecyclerView rvNewsFeed = v.findViewById(R.id.rv_news_feed);
```

Comment:
you should service to get the news feed , and mapper if needed

Reply:
The Firebase is a AsyncTask, It works in the background anyway.
So i can see little advantage to that especially because the OS will close our application anyway which means the service would be an over-engineered solution. 
So what do u think? 

About the mapper i can see the point, i used it in my last push, but i found it more functional to map this way, however i think your approach is useful for architecture purposes